### ISSUE TITLE
ZIO.never zipPar ZIO.never becomes uninterruptible after garbage collection

### ISSUE THREAD

**Issue by Marc-Ducret** [zio/zio#7251]
```scala
import zio._

object TestZIOPar extends ZIOAppDefault {
  override def run: ZIO[ZIOAppArgs with Scope, Nothing, Unit] =
    ZIO.debug("Started").delay(1.second) *> {
      ZIO.debug("GC will run") *> ZIO.succeed(java.lang.Runtime.getRuntime.gc()) <* ZIO.debug("GC ran")
    }.delay(1.second).fork *>
      (ZIO.never <&> ZIO.never).unit
}
```
This program can be stopped with an interruption signal between the printing of `Started` and `GC will run`.
However, after `GC ran` is printed, the program cannot be stopped using an interruption signal.

In practice this behavior can happen relatively easily when running multiple web servers in parallel (many libraries use `ZIO.never` in the effect running the web server).

**Comment by adamgfraser** [comment]
@Marc-Ducret I am not able to reproduce on `series/2.x`.

**Comment by Marc-Ducret** [comment]
I have tested with:
```
scala: 2.13.8
zio:   2.0.1 and 2.0.1+12-4cbf2640-SNAPSHOT
java:  1.11.0-openjdk-amd64
OS:    Ubuntu 20.04.4
```
And somehow it works as described every time for me
![image](https://user-images.githubusercontent.com/2799996/186706570-6c924e58-a2a3-4bb7-9d44-84fa5e0cc6db.png)
![image](https://user-images.githubusercontent.com/2799996/186706705-5f5609ae-6f0c-42d5-8787-7149319b161e.png)

build.sbt
```scala
ThisBuild / scalaVersion := "2.13.8"

lazy val root = (project in file("."))
  .settings(
    name := "ZIO-Stream-Map-ZIO-Par",
    resolvers += "Sonatype OSS Snapshots" at "https://oss.sonatype.org/content/repositories/snapshots",
    libraryDependencies += "dev.zio" %% "zio" % "2.0.1+12-4cbf2640-SNAPSHOT",
    libraryDependencies += "dev.zio" %% "zio-streams" % "2.0.1+12-4cbf2640-SNAPSHOT",
    libraryDependencies += "dev.zio" %% "zio-test" % "2.0.1+12-4cbf2640-SNAPSHOT"
  )
```

Edit:
It also does as described when running with `sbt run TestZIOPar` and using `Ctrl + C` to interrupt the program.
And I also tested with java 1.8 / 1.17 and/or scala 3.1.2.

**Comment by adamgfraser** [comment]
Minimized to just use ZIO:

```scala
object Example extends ZIOAppDefault {

  val workflow =
    ZIO.debug("Started").delay(1.second) *> {
        ZIO.debug("GC will run") *> ZIO.succeed(java.lang.Runtime.getRuntime.gc()) <* ZIO.debug("GC ran")
      }.delay(1.second).fork *>
         (ZIO.never <&> ZIO.never).unit

  val run =
    for {
      fiber <- workflow.forkDaemon
      _     <- ZIO.sleep(5.seconds)
      _     <- ZIO.debug("interrupting")
      _     <- fiber.interrupt.debug
      _     <- ZIO.debug("interrupted")
    } yield ()
}
```

**Comment by danielleontiev** [comment]
Is there a chance that the fix will be backported to the stable `2.0.x` branch? I experience this issue on `zio` `2.0.21`
